doc/scenario.md

```diff
@@ -0,0 +1,439 @@
+you can add `special_magic` step to scenario. Just pass path to a directory where your custom steps are placed:
+
+```yaml
+- name: My playbook
+  hosts: all
+  tasks:
+    - name: Deploy application
+      import_role:
+        name: tarantool.cartridge
+      vars:
+        cartridge_custom_steps_dir: "./custom_steps"
+        cartridge_scenario:
+          - restart_instance
+          - special_magic
+```
+
+### Importing steps from different directories
+
+If you want to import steps by full paths, `cartridge_custom_steps` parameter can be useful. It allows specifying
+mapping between step name and path to file to import:
+
+```yaml
+cartridge_custom_steps:
+  - name: 'common_special_task'
+    file: './common/special_task.yml'
+```
+
+### Override task from default scenario
+
+Sometimes we need to use default scenario, but change some steps realizations.
+
+For example, sometimes we need to deliver package to remote machine by our own way. By default, package specified
+by `cartridge_package_path` is simple copied to remote host, it's happened on `deliver_package` step. Sometimes we need
+to download or copy package from some custom place. Let's do it!
+
+Generally, we just need to override default `deliver_package` step.
+
+Let's take a look on it's [API](#deliver_package). It requires two variables:
+
+- `cartridge_package_path` that we simply specify in vars;
+- `single_instances_for_each_machine` that allows us to run this task once per each machine.
+
+As a result of this module we should set `delivered_package_path` variable
+(a path of package on remote machine).
+
+Now, choose a directory where our custom steps are placed, for example `./custom_steps`.
+
+Then create `./custom_steps/deliver_package.yml` and describe a way to deliver package:
+
+```yaml
+# ./custom_steps/deliver_package.yml
+
+- when:
+    - cartridge_package_path is not none
+    - inventory_hostname in single_instances_for_each_machine
+  tags: cartridge-instances
+  block:
+    - name: 'Download package'
+      get_url:
+        url: 'https://some-server/myapp-1.0.0-0.rpm'
+        dest: '/tmp'
+      register: downloaded_package
+
+    - name: 'Set delivered package path'
+      set_fact:
+        delivered_package_path: '{{ downloaded_package.dest }}'
+```
+
+Import role and say where to find our custom steps:
+
+```yaml
+# deploy_application.yml
+
+- name: My playbook
+  hosts: all
+  tasks:
+    - name: Deploy application
+      import_role:
+        name: tarantool.cartridge
+      vars:
+        cartridge_custom_steps_dir: "./custom_steps"
+```
+
+### Editing topology without connecting to membership
+
+Now there is a big problem on deploying huge clusters - `connect_to_membership` step is too long. Using scenario, we can
+solve this problem until it isn't solved in `cartridge`.
+
+In fact `connect_to_membership` is used to [find some instance that is already in cluster](#connect_to_membership). This
```

Comment:
Why link to `connect_to_membership` is here?   It can be at the previous line:  `...[connect_to_membership](#link) step is too long.`